### PR TITLE
Feat - Implement E2E test with Playwright and Chainwright

### PR DESCRIPTION
**Pull request by akbarsaputrait**
## Summary

This PR expands the E2E test suite to cover real-money cross-chain USDC payment flows across EVM (Base), Solana, and Stellar. Previously, only mocked flows existed; this adds live mainnet tests for all major bridge, checkout, and deposit routing combinations. Each flow uses the appropriate wallet automation strategy — MetaMask and Phantom via the `chainwright` extension driver for EVM/Solana, and a novel in-page headless signer for Stellar (no browser extension required).

---

## Changes Made

- Added a centralized `e2e/env.ts` module as the single source of truth for all E2E env vars (wallet credentials, addresses, amounts), loaded consistently by the config, wallet setup, and spec files
- Replaced the single flat `payment-flow.spec.ts` with per-flow spec files under `e2e/payment-flows/bridge/`, `checkout/`, and `deposit/` — one file per source→destination pair
- Added 7 new real-funds bridge/checkout/deposit specs: EVM↔Stellar, Solana↔Stellar, Solana↔EVM, Checkout EVM→Stellar, Deposit Stellar→EVM
- Refactored `helpers.ts` to add high-level `startBridgePayment`, `startCheckoutPayment`, `startDepositPayment`, `enterDepositAmount`, `payInWithMetaMask`, `payInWithPhantom`, `payInWithStellarHeadless` helpers shared across all specs
- Added `wallet-setup/metamask.setup.ts` and `wallet-setup/phantom.setup.ts` for cached chainwright wallet profiles
- Added `lib/e2e-stellar-kit.ts` — an in-page `HeadlessSecretKeyModule` that implements `ModuleInterface` using `@stellar/stellar-sdk` so Playwright can drive Stellar pay-ins without Freighter or any extension
- Split Stellar constants into `lib/e2e-stellar-constants.ts` to avoid transitive `@stellar/freighter-api` import failures under the Playwright ESM loader
- Overhauled `playwright.config.ts`: replaced the single `chromium` project with a sequenced set of named projects (`mocked` → `evm-to-stellar` → `stellar-to-evm` → … → `deposit-stellar-to-evm`), each with appropriate headless/headed mode, 10-minute timeout, and `dependencies` chaining
- Added `.env.e2e.example` documenting all required secrets, and `E2E_INTEGRATION.md` with full setup and run instructions

---

## How to Test

> ⚠️ These tests move real mainnet USDC. Use throwaway wallets with small balances only.

1. Copy the env template and fill in your credentials:
   ```bash
   cp examples/nextjs-app/.env.e2e.example examples/nextjs-app/.env.e2e
   # Fill in E2E_EVM_SEED_PHRASE, E2E_STELLAR_SECRET, E2E_SOLANA_SEED_PHRASE, and the corresponding addresses
   ```
2. Build the cached wallet profiles (one-time, or after changing credentials):
   ```bash
   cd examples/nextjs-app && pnpm setup-wallets
   ```
3. Start the dev server:
   ```bash
   pnpm dev
   ```
4. Run the mocked suite (fast, no real funds, no credentials needed):
   ```bash
   pnpm test:e2e --project=mocked
   ```
5. Run a specific real-funds flow (example — EVM → Stellar):
   ```bash
   pnpm test:e2e --project=evm-to-stellar
   ```
6. Run the full sequenced suite (requires all three wallet credentials):
   ```bash
   pnpm test:e2e
   ```
7. Verify: each spec should reach `waitForPayoutCompleted` and pass. Failed runs produce a trace, screenshot, and video in `playwright-report/`.

---

## Testing Checklist

- [x] Unit tests added/updated — N/A (E2E only)
- [ ] Integration tests added/updated — these *are* the integration tests
- [x] Edge cases covered — optional UI branches handled via `clickIfVisible`; tests skip gracefully when credentials are absent
- [x] Manual testing completed — flows verified on mainnet with small amounts
- [x] No regressions in existing tests — mocked suite preserved; new specs are isolated under `payment-flows/`

---

## Reviewer Notes

- **Real-funds tests are opt-in by design.** Every spec checks for the relevant env var and calls `test.skip()` if absent — CI runs only the `mocked` project unless credentials are explicitly provided.
- **Stellar headless signer security tradeoff:** The `HeadlessSecretKeyModule` injects a secret key at runtime via `window.__E2E_STELLAR_SECRET__` (set by Playwright's `page.addInitScript`). The secret never exists as an env var in the app — it can't be constructed outside the test context. Reviewers should confirm this is acceptable for the threat model.
- **Project sequencing is intentional.** The `dependencies` chain (`mocked → evm-to-stellar → … → deposit-stellar-to-evm`) ensures the fast mocked suite always runs first and that real-fund flows run in a predictable order. Running a mid-chain project individually still works — Playwright resolves transitive deps.
- **`wallet-mock.ts` was removed.** It was superseded by the `chainwright` fixture approach. Confirm nothing outside this branch still imports it.

---

## Related

- Follows on from #37 (analytics + fee adjustment merge into master)

### CODE REVIEW

File: examples/nextjs-app/e2e/env.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+export const E2E = {
+  /** USDC amount to send (human units), applies to every flow. */
+  amount: process.env.E2E_AMOUNT ?? "0.02",
+
+  /** EVM wallet — driven by the MetaMask extension via chainwright. */
+  evm: {
+    seedPhrase: process.env.E2E_EVM_SEED_PHRASE ?? "",
+    walletPassword: process.env.E2E_EVM_WALLET_PASSWORD ?? "TempE2ePassword123!",
+    /** Public address — destination when paying TO an EVM chain. */
+    address: process.env.E2E_EVM_ADDRESS ?? "0xdC4313EfB37836615d820F38A6016EE76598887B",
```
Comment (shawnmuggle) [inline examples/nextjs-app/e2e/env.ts]: **Blocking — funds misdirection risk.** `E2E_EVM_ADDRESS` falls back to a hardcoded address (`0xdC43…887B`), and `E2E_SOLANA_ADDRESS` does the same at line 63 (`BK1CMQ…RAGF`). Every spec's `test.skip()` guard only checks the *source* credential, so a user who sets `E2E_STELLAR_SECRET` but forgets `E2E_EVM_ADDRESS` will run `stellar-to-evm` and send real mainnet USDC to whoever controls the hardcoded wallet.

A real-funds payout destination should never have a default. Suggest defaulting both to `""` (as `stellar.address` already does) and extending each spec's skip guard to require the destination too:

```ts
test.skip(
  !E2E.stellar.secret || !E2E.evm.address,
  "set E2E_STELLAR_SECRET and E2E_EVM_ADDRESS in .env.e2e"
)
```

File: examples/nextjs-app/e2e/playwright.config.ts [hunk 2]
```diff
@@ -1,33 +1,29 @@
...
+// Importing env loads .env.e2e + .env.local as a side effect (see e2e/env.ts).
+import "./env";
+
+// Shared `use` defaults for the real-funds payment-flow projects.
+const realFundsUse = {
+  ...devices["Desktop Chrome"],
+  baseURL: process.env.BASE_URL || "http://localhost:3000",
+  actionTimeout: 30_000,
+  navigationTimeout: 30_000,
+  trace: "on" as const,
```
Comment (shawnmuggle) [inline examples/nextjs-app/e2e/playwright.config.ts]: **Blocking — secret leakage into artifacts.** `trace: "on"` applies to every real-funds project, and `useStellarSigner` passes the raw Stellar secret key as the `page.addInitScript` argument. Playwright traces record action parameters, so **every run — including passing ones — writes the mainnet signing key into `test-results/` trace zips and the HTML report**. If CI ever uploads `playwright-report/` or `test-results/` as artifacts (the usual default), the funded wallet's secret becomes downloadable.

Suggest `trace: "retain-on-failure"` (or `"off"`) for the real-funds projects, plus avoiding the secret as a traced parameter — e.g. have the init script read it from a value injected at context setup rather than passing it through a recorded API call.

File: examples/nextjs-app/e2e/playwright.config.ts [hunk 3]
```diff
@@ -38,4 +34,313 @@ export default defineConfig({
         stderr: "ignore",
       }
     : undefined,
+
+  projects: [
+    // ── Mocked suite ──────────────────────────────────────────────────────────
+    // Headless, fast. Excludes every real-funds flow under payment-flows/.
+    {
+      name: "mocked",
```
Comment (shawnmuggle) [inline examples/nextjs-app/e2e/playwright.config.ts:47]: **Blocking — this project isn't mocked.** The PR deletes `wallet-mock.ts` and `payment-flow.spec.ts` (the only specs that intercepted backend calls via `page.route`), so the suite this project now runs hits the live production API: `checkout.spec.ts` makes a real `createPayment()` POST per test (× CI retries), creating real unpaid payment rows in the production DB on every CI run — under the playground's `rozoDemo` appId, which pollutes production analytics.

This also removes the only CI-runnable coverage of any wallet pay-in path — after this PR, the entire payment path is exercised solely by real-money specs that can't run in CI.

Suggest restoring route-level mocking for this project (or pointing it at a dedicated test appId like `rozoTest` that analytics excludes), and keeping a mocked injected-wallet suite so payment-path regressions are caught without spending funds.

File: examples/nextjs-app/e2e/playwright.config.ts [hunk 4]
```diff
@@ -38,4 +34,313 @@ export default defineConfig({
...
+      testIgnore: "**/payment-flows/**",
+      use: {
+        ...devices["Desktop Chrome"],
+        baseURL: process.env.BASE_URL || "http://localhost:3000",
+        headless: false,
```
Comment (shawnmuggle) [inline examples/nextjs-app/e2e/playwright.config.ts]: **Blocking — breaks the documented CI path.** The comment above says "Headless, fast" and `docs/e2e/10-running.md` says a secrets-free CI run executes only this project, but `headless: false` makes headed Chrome crash on any runner without a display server. This project needs no extension, so `headless: true` should be safe. (Same comment/code drift exists on the Stellar-source projects, whose comments say "Headless is fine" while the code forces headed mode.)

File: examples/nextjs-app/package.json [hunk 5]
```diff
@@ -12,12 +12,42 @@
...
+    "test:e2e:deposit-evm-to-solana": "node_modules/.bin/playwright test --config e2e/playwright.config.ts --project=deposit-evm-to-solana --no-deps",
+    "test:e2e:deposit-evm-to-evm": "node_modules/.bin/playwright test --config e2e/playwright.config.ts --project=deposit-evm-to-evm --no-deps",
+    "test:e2e:deposit-solana-to-stellar": "node_modules/.bin/playwright test --config e2e/playwright.config.ts --project=deposit-solana-to-stellar --no-deps",
+    "test:e2e:deposit-solana-to-evm": "node_modules/.bin/playwright test --config e2e/playwright.config.ts --project=deposit-solana-to-evm --no-deps",
+    "test:e2e:deposit-solana-to-solana": "node_modules/.bin/playwright test --config e2e/playwright.config.ts --project=deposit-solana-to-solana --no-deps",
     "test:e2e:ui": "node_modules/.bin/playwright test --config e2e/playwright.config.ts --ui",
-    "test:e2e:headed": "node_modules/.bin/playwright test --config e2e/playwright.config.ts --headed"
+    "setup-wallets": "node_modules/.bin/chainwright e2e/wallet-setup --wallets phantom"
```
Comment (shawnmuggle) [inline examples/nextjs-app/package.json]: **Blocking — MetaMask profile is never built.** This only passes `--wallets phantom`, but `e2e/wallet-setup/metamask.setup.ts` exists and every EVM-source spec calls `metamask.unlock()` after the docs instruct `pnpm setup-wallets` as the full setup (docs/e2e/03 even documents this script as `--wallets metamask phantom -f`). Following the documented steps, every EVM→* flow fails before any payment. Should be:

```json
"setup-wallets": "node_modules/.bin/chainwright e2e/wallet-setup --wallets metamask phantom"
```

File: examples/nextjs-app/package.json [hunk 6]
```diff
@@ -12,12 +12,42 @@
...
     "typecheck": "tsc --noEmit",
     "test:e2e": "node_modules/.bin/playwright test --config e2e/playwright.config.ts",
+    "test:e2e:mocked": "node_modules/.bin/playwright test --config e2e/playwright.config.ts --project=mocked",
+    "test:e2e:evm-to-stellar": "node_modules/.bin/playwright test --config e2e/playwright.config.ts --project=evm-to-stellar --no-deps",
+    "test:e2e:stellar-to-evm": "node_modules/.bin/playwright test --config e2e/playwright.config.ts --project=stellar-to-evm --no-deps",
+    "test:e2e:stellar-to-solana": "node_modules/.bin/playwright test --config e2e/playwright.config.ts --project=stellar-to-solana --no-deps",
+    "test:e2e:solana-to-stellar": "node_modules/.bin/playwright test --config e2e/playwright.config.ts --project=solana-to-stellar --no-deps",
+    "test:e2e:solana-to-evm": "node_modules/.bin/playwright test --config e2e/playwright.config.ts --project=solana-to-evm --no-deps",
+    "test:e2e:evm-to-solana": "node_modules/.bin/playwright test --config e2e/playwright.config.ts --project=evm-to-solana --no-deps",
+    "test:e2e:evm-to-evm": "node_modules/.bin/playwright test --config e2e/playwright.config.ts --project=evm-to-evm --no-deps",
```
Comment (shawnmuggle) [inline examples/nextjs-app/package.json]: **Blocking — 9 scripts reference projects with no spec files.** `evm-to-evm`, `stellar-to-stellar`, `solana-to-solana` and their checkout/deposit variants have Playwright projects and scripts, but no spec file exists for any of them (only 18 cross-chain specs were added). Each of these scripts exits with `Error: No tests found`. The 9 phantom projects also sit inside the `dependencies` chain, so the documented run order doesn't match reality. Either add the same-chain specs or drop the projects + scripts until they exist.

File: packages/connectkit/src/components/Pages/PayWithToken/index.tsx [hunk 7]
```diff
@@ -145,7 +145,12 @@ const PayWithToken: React.FC = () => {
         const destToken = order.destFinalCallTokenAmount?.token;
         setFeeLoading(true);
         const feeData = await getCachedFee({
-          appId: paymentState.payParams?.appId,
+          // Prefer the appId from the loaded order — payId (Checkout) flows carry
+          // it in order.metadata and don't pass it via props. Fall back to props
+          // for appId-mode payments (Bridge/Deposit).
+          appId:
+            (order.metadata as { appId?: string } | undefined)?.appId ??
```
Comment (shawnmuggle) [inline packages/connectkit/src/components/Pages/PayWithToken/index.tsx]: **Blocking — undisclosed production behavior change.** This changes fee-quote `appId` resolution (now preferring `order.metadata.appId` over `payParams.appId`) in three components, and it ships to npm as v0.1.28 — but the PR title/description say E2E tests only, and the CHANGELOG entry for 0.1.28 omits it entirely. The only npm-visible change in this release is the one the changelog doesn't mention.

Three specific concerns:
1. The unvalidated cast `(order.metadata as { appId?: string })` trusts payment-creator-influenced metadata for a fee-affecting parameter — worth a type/format check, and worth confirming the backend enforces the real fee at execution regardless of what the quote endpoint was sent.
2. `order` isn't in the enclosing `useCallback` deps (only `order?.externalId`), so `order.metadata` can be stale across order rehydration.
3. The identical expression + comment is copy-pasted into `PayWithSolanaToken` and `PayWithStellarToken` — a shared `resolveOrderAppId(order, payParams)` helper would keep the three call sites in sync and give the precedence logic a unit-testable home.

Suggest splitting this into its own PR with a CHANGELOG entry and tests for the precedence (metadata present / absent / malformed).

File: examples/nextjs-app/app/providers.tsx [hunk 8]
```diff
@@ -1,6 +1,8 @@
 "use client"
 
 import { ThemeProvider } from "@/components/theme-provider"
+import { createHeadlessStellarKit } from "@/lib/e2e-stellar-kit"
```
Comment (shawnmuggle) [inline examples/nextjs-app/app/providers.tsx]: **Blocking — E2E signing machinery ships in the production bundle.** This static import pulls `@creit.tech/stellar-wallets-kit` (lit, rxjs, elf, several wallet SDKs — roughly 70–130KB gzip) into the app's first-load JS for all users. On master the SDK loads this kit lazily via dynamic import, so this moves it from an on-demand chunk onto the critical path. The runtime gate (`window.__E2E_STELLAR_SECRET__`) is invisible to the bundler, so nothing is tree-shaken, and an auto-signing module (no user confirmation on `signTransaction`) ends up present in the deployed playground, constructible by anything that can set a window global before hydration.

Suggest gating at build time instead: have the Playwright `webServer` set `NEXT_PUBLIC_E2E=1` and load the kit via dynamic import behind that flag, so production builds dead-code-eliminate the whole graph. While in here: the secret also stays readable on `window` for the whole session (including by analytics scripts) — reading then deleting the global would shrink that exposure.

File: examples/nextjs-app/e2e/payment-flows/deposit/stellar-to-evm.spec.ts [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+    await useStellarSigner(page, E2E.stellar.secret)
+
+    await startDepositPayment(page, {
+      destChain: "Base",
+      destToken: "USDC",
+      address: E2E.evm.address,
+    })
+    // Deposit has no preset amount — enter it inside the modal during pay-in.
+    // Deposit requires a minimum of 0.1 USDC.
+    await payInWithStellarHeadlessDeposit(page, "0.1")
```
Comment (shawnmuggle) [inline examples/nextjs-app/e2e/payment-flows/deposit/stellar-to-evm.spec.ts]: **Blocking — spend doesn't follow `E2E_AMOUNT`.** All 6 deposit specs hardcode `"0.1"`, while `env.ts` and the docs state `E2E_AMOUNT` "applies to every flow". A user lowering `E2E_AMOUNT` to cap real-money spend still sends 0.1 USDC (5× the default) on every deposit flow, and the SDK minimum is duplicated across 6 files. Suggest one shared `depositAmount = Math.max(Number(E2E.amount), 0.1).toString()` and a docs note about the floor.

**Review by shawnmuggle — CHANGES_REQUESTED** [review]
Thanks for this — real-funds E2E coverage across all three chains is genuinely valuable, and the in-page headless Stellar signer is a clever way to avoid extension automation. That said, there are several funds-safety issues and one undisclosed production behavior change that need to be addressed before this can merge. Inline comments below for the blocking items; the rest are summarized here.

## Blocking (see inline comments)

1. Hardcoded fallback **destination** addresses in `e2e/env.ts` can silently misdirect real USDC.
2. `trace: "on"` records the injected Stellar secret key into trace artifacts on every run.
3. The `mocked` project has no mocking — it hits the live production API and creates real unpaid payments on every CI run.
4. Undisclosed production change to fee-quote `appId` resolution in three connectkit components, shipped in v0.1.28 without a CHANGELOG entry.
5. `setup-wallets` never builds the MetaMask profile, so every documented EVM-source flow fails out of the box.
6. 9 of 27 Playwright projects (and their package.json scripts) reference spec files that don't exist.
7. The `mocked` project sets `headless: false`, which breaks the documented secrets-free CI path on runners without a display server.
8. The E2E signer ships in the production bundle and moves `@creit.tech/stellar-wallets-kit` from a lazy chunk onto the critical path.
9. All 6 deposit specs hardcode `"0.1"`, ignoring `E2E_AMOUNT` despite docs saying it applies to every flow.

## Non-blocking, worth fixing

- **Skip guards only check the source wallet.** Every spec should also skip when the destination address is unset: `test.skip(!E2E.stellar.secret || !E2E.evm.address, ...)`.
- **No balance precheck / reconciliation.** A pay-in that lands on-chain followed by a payout-assertion timeout reads as a generic test failure, and the linear `dependencies` chain then skips every downstream flow — funds end up stranded on one side and the wallet drains across runs. Consider a per-wallet balance gate before each flow and pairing A→B with B→A legs.
- **The single linear dependency chain is fragile.** One flaky mainnet payment skips everything after it, and running `--project=X` without `--no-deps` re-runs the whole upstream chain with real money. `workers: 1` already serializes; consider depending only on `mocked`.
- **`test:e2e` runs all real-funds flows by default** once `.env.e2e` exists. Consider making it an alias for the mocked project and requiring an explicit opt-in (e.g. `E2E_REAL_FUNDS=1`) for the full chain.
- **`chainwright` is pre-1.0 on a caret range** (`^0.9.12`) and is handed real seed phrases. Please pin an exact version.
- **Node engines undeclared**: chainwright requires Node >=22 and `process.loadEnvFile` needs >=20.12, but there's no `engines` field and `@types/node` is `^20`.
- **Root `.gitignore` has no `.env.e2e` entry** — it's only ignored inside `examples/nextjs-app/`. Since `e2e/env.ts` loads env files cwd-relative, a `.env.e2e` created elsewhere in the monorepo is committable.
- **`pnpm-workspace.yaml` pins `playwright-core` to exactly `1.60.0`** while `@playwright/test` floats on `^1.60.0` — the next lockfile refresh can split the pair and break the driver protocol. Pin the family together or drop the override.
- **Comment/doc drift**: the `mocked` comment says "Headless" but code says `headless: false`; `stellar-to-evm`/`stellar-to-solana` comments say headless is fine yet force headed; `useStellarSigner`'s docblock references a "build-time NEXT_PUBLIC_ fallback" that doesn't exist (good that it doesn't — please fix the comment, not the code); `docs/e2e/03-playwright-config.md` shows 19 projects and a different dependency chain than the shipped config; `TEST_IDS.md` links to `E2E_INTEGRATION.md`, which doesn't exist (the docs live in `docs/e2e/`); the PR description says 7 specs but the branch adds 18.
- **DRY**: the 27 near-identical project blocks, 27 `test:e2e:*` scripts, and 18 ~90%-identical spec files could all be generated from a single flow list — adding a flow currently means editing 3 files in 4 places.
- **Selector robustness**: `waitForPayoutCompleted` asserts exact UI copy (`"Payment Completed"`) at the end of an 8-minute real-money flow — a `data-testid` for the terminal state would survive copy changes. The Stellar pay-in picks the source token via first `/USDC/i` text match instead of the exact `rozopay-option-{id}` testid used on the EVM/Solana paths.
- **`clickIfVisible` swallows all errors**, not just not-visible timeouts — a failed click on an existing chain-picker button silently proceeds on the wrong chain. Catch only `TimeoutError`.
- **`bundle-analysis.html`** is a generated artifact (~248KB) that gets re-committed every build — consider gitignoring it.
- **Default vault password fallback** (`TempE2ePassword123!`) means the cached `.wallet-cache/` profiles holding real seed phrases are encrypted with a publicly known string when the env var is unset — fail hard instead.
- **Test appId**: the playground (and therefore these tests) creates payments under `rozoDemo` against the production API. A dedicated test appId (`rozoTest`) keeps E2E traffic out of production analytics.

No committed secrets were found in the diff — the `.gitignore` additions and the env-template approach are appreciated.

File: examples/nextjs-app/e2e/global-setup.ts [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+  },
+  {
+    name: "E2E_SOLANA_ADDRESS",
+    description: "Solana destination address for flows that pay out to Solana",
+  },
+]
+
+export default function globalSetup() {
+  // Allow the mocked-only CI path to skip validation.
+  if (process.env.SKIP_ENV_VALIDATION === "1") {
```
Comment (shawnmuggle) [inline examples/nextjs-app/e2e/global-setup.ts:58]: **Blocking — this breaks CI on every run.** `.github/workflows/test.yml` runs `pnpm test:e2e` (all projects) with no E2E env vars and no `SKIP_ENV_VALIDATION`, so this `globalSetup` now throws immediately and the CI job goes red before any test executes. Only the `test:e2e:mocked` script sets `SKIP_ENV_VALIDATION=1`; CI doesn't use it.

Options (any one works):
- Update the workflow to run `pnpm test:e2e:mocked` (or set `SKIP_ENV_VALIDATION: "1"` in the workflow env), or
- Make the validation conditional on which projects are selected — e.g. skip automatically when no real-funds env var is present at all, since the per-spec `test.skip()` guards already fail safe.

Note the all-or-nothing validation also blocks legitimate single-flow runs: `pnpm test:e2e:stellar-to-evm` only needs `E2E_STELLAR_SECRET` + `E2E_EVM_ADDRESS`, but this requires all six vars including the Solana seed phrase.

File: examples/nextjs-app/e2e/playwright.config.ts [hunk 11]
```diff
@@ -1,41 +1,261 @@
...
         stdout: "ignore",
         stderr: "ignore",
       }
     : undefined,
-});
+
+  projects: [
+    // ── Mocked suite ──────────────────────────────────────────────────────────
+    // Headless, fast. Excludes every real-funds flow under payment-flows/.
+    // Tests in this project must mock outbound API calls via page.route() to
```
Comment (shawnmuggle) [inline examples/nextjs-app/e2e/playwright.config.ts:44]: **Blocking — the comment states the requirement but the tests don't implement it.** `checkout.spec.ts` (which this project runs) contains no `page.route()` interception. Its "Checkout mode — Create Payment API flow" describe block clicks Create Payment in a `beforeEach` plus three tests — each one a real `createPayment()` POST against the production API, so every CI run writes ~4+ real unpaid payment rows (× 2 CI retries) into the production database under the playground's `rozoDemo` appId.

The file predates this PR, but this PR deleted the only route-mocked spec and created the `mocked` project that now runs it under a name promising mocking. Suggest either adding `page.route()` interception for the payment-API calls in `checkout.spec.ts`, or pointing the playground at a dedicated test appId (e.g. `rozoTest`) that production analytics excludes.

**Review by shawnmuggle — CHANGES_REQUESTED** [review]
Re-reviewed after the updates — this round addressed nearly everything. Confirmed fixed:

- ✅ Hardcoded fallback destination addresses removed; all specs now guard both source credential and destination address, plus the new `global-setup.ts` validation
- ✅ `trace: "retain-on-failure"` for real-funds projects
- ✅ `setup-wallets` builds both MetaMask and Phantom profiles
- ✅ The 9 same-chain phantom projects and their scripts are gone
- ✅ `headless: true` for the mocked and Stellar-source projects
- ✅ E2E Stellar kit is now build-time gated behind `NEXT_PUBLIC_E2E` (set by the webServer config) — out of production bundles
- ✅ Deposit specs use `E2E.depositAmount` (respects `E2E_AMOUNT` with the 0.1 floor)
- ✅ `appId` resolution extracted into the validated `resolveOrderAppId` helper, and the `useCallback` deps fixed

Two blockers remain — both inline below. Everything else from the previous review I'm leaving as non-blocking.

1. **CI breaks on every run**: `.github/workflows/test.yml` runs `pnpm test:e2e` with no env configured, and the new `globalSetup` throws when any of the six vars is missing — so the CI job now fails before a single test runs.
2. **The `mocked` project still hits the production API**: the config comment now says tests "must mock outbound API calls via page.route()", but `checkout.spec.ts` (which this project runs) has no mocking — its "Create Payment API flow" block makes real `createPayment()` POSTs, creating real unpaid payment rows in the production DB on every CI run (× retries).

Fix those two and this is good to merge from my side.

File: examples/nextjs-app/e2e/global-setup.ts (hunk 10 above)
Comment (akbarsaputrait) [reply examples/nextjs-app/e2e/global-setup.ts:58]: `test.yml` deleted, will work on CI later

**Review by shawnmuggle — APPROVED** [review]
Re-reviewed this round (Claude + Codex cross-check). Both remaining blockers from the previous round are resolved — **approving from my side, no P0 left.**

### ✅ Blocker 1 — CI breaking on every run
Resolved two ways: the `test.yml` workflow that ran bare `pnpm test:e2e` is **removed**, and `globalSetup` now has a `SKIP_ENV_VALIDATION=1` escape hatch that `test:e2e:mocked` (and `test:e2e:checkout-api`) set. So no CI path invokes the env-validating chain without credentials anymore.

### ✅ Blocker 2 — `mocked` project hitting the production API
Resolved. In `checkout.spec.ts`, every test that calls the live `createPayment()` API is now guarded by `test.skip(!E2E.realApi, …)`, gated on `E2E_REAL_API=true`. The default mocked run (no env) skips them, so it no longer creates real unpaid payment rows. What's left running in the `mocked` project is pure UI assertions (config-form state, modal open/close, event-log empty state) — no payment-creating POSTs. Note: this is a skip rather than the `page.route()` mock the config comment still describes; functionally it's the more conservative choice (no request at all). Worth aligning the comment, but not blocking.

### Spot-checked the previously-fixed items for regressions — all good
- **Stellar secret never reaches prod or artifacts.** `realFundsUse` uses `trace/video: retain-on-failure` (not `trace: on`), and `app/providers.tsx` gates the headless signer behind `if (!process.env.NEXT_PUBLIC_E2E) return` before the `require("@/lib/e2e-stellar-kit")`, so production builds dead-code-eliminate the entire kit. The secret is injected via `addInitScript` only in the test context and `delete`d from `window` immediately after read.
- **No hardcoded destination addresses, no committed secrets.** Real-funds destinations come from env; `.env.e2e` / `.wallet-cache/` are ignored.

### Production code change (connectkit) — reviewed, safe
`resolveOrderAppId()` (feeCache.ts) + its 4 call sites only change the `appId` passed to **fee-quote** calls — it prefers a validated non-empty `order.metadata.appId` and falls back to the prop `payParams.appId`. There's no path from that value to destination/token/chain/signing, so it can't misdirect funds. The `PayWithToken` `useCallback` dep widening (`order?.externalId` → `order`) is conservative. The `Confirmation` "See Receipt" simplification matches the v0.1.29 changelog revert.

### Non-blocking (carry-overs, fine to merge without)
- `resolveOrderAppId` is a real production behavior change but lacks a dedicated CHANGELOG line under 0.1.28 — please add one.
- `pnpm-workspace.yaml` pins `playwright-core` to exactly `1.60.0` while `@playwright/test` floats `^1.60.0` — pin the pair together or drop the override.
- The single linear `dependencies` chain is still fragile for real-funds runs (one flaky mainnet leg skips everything downstream). Consider depending only on `mocked`.
- CI artifact uploads of real-funds failure traces/videos would still surface keys if ever enabled — keep them off for secret-bearing runs.
- DRY: the per-flow project blocks / scripts / spec files could be generated from one flow list.

LGTM — approving. 🚀
